src/cachemachine/rubinimageinfo.py

```diff
@@ -0,0 +1,346 @@
+import re
+from dataclasses import dataclass
+from enum import Enum, auto
+from typing import Optional
+
+import semver
+from docker_image import reference
+
+from .types import IncomparableImagesError
+
+DEFAULT_TAG = "latest"  # Docker convention
+DEFAULT_RECOMMENDED_TAG = "recommended"
+
+
+class RubinImageType(Enum):
+    """Class to capture the different kinds of image types possible in a
+    Rubin Lab Image container, given our standard tag set."""
+
+    RECOMMENDED = auto()
+    LATEST = auto()
```

Comment:
Since we have recommended, we shouldn't probably worry about latest.

Reply:
I can treat latest and latest_something as malformed for now, but T&S is cheerily building latest, latest_weekly, and latest_daily.  Ultimately we should probably treat them like recommended, where we scan and match digests.